src/Illuminate/Redis/Connections/PhpRedisConnection.php

```diff
@@ -55,10 +55,18 @@ public function mget(array $keys)
      */
     public function set($key, $value, $expireResolution = null, $expireTTL = null, $flag = null)
     {
+        $extendedOptions = null;
+        if ($expireResolution) {
+            $extendedOptions = [$expireResolution];
+        }
+        if ($expireTTL && $flag) {
+            $extendedOptions[$flag] = $expireTTL;
+        }
+
```

Comment:
I find this hard to read/comprehend and it's hard to make changes in the future to this without tests.